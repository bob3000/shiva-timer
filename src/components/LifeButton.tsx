import * as React from 'react';
import {
  ImageBackground,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  View,
  ViewStyle,
} from 'react-native';

interface ILifeButtonProps {
  background?: ViewStyle;
  onPress?: () => void;
  size?: number | string;
  title?: string;
  textSize?: number;
}

interface ILifeButtonStyle {
  background: ViewStyle;
  mainContainer: ViewStyle;
  titleText: TextStyle;
  titleTextContainer: ViewStyle;
}

export const LifeButton: React.FunctionComponent<ILifeButtonProps> = (
  props,
) => {
  const styles = StyleSheet.create<ILifeButtonStyle>({
    background: {
      opacity: 0.8,
      ...props.background,
    },
    mainContainer: {
      alignItems: 'center',
      borderColor: '#000000',
      borderRadius: 100,
      borderWidth: 1,
      height: props.size,
      justifyContent: 'center',
      opacity: 0.8,
      width: props.size,
    },
    titleText: {
      color: '#000000',
      fontSize: props.textSize,
      fontWeight: 'bold',
    },
    titleTextContainer: {
      borderRadius: 40,
      opacity: 1,
      padding: 5,
    },
  });

  return (
    <TouchableOpacity activeOpacity={0.7} onPress={props.onPress}>
      <ImageBackground
        source={require('../../assets/images/flower_of_life.png')}
        style={[styles.background, styles.mainContainer]}
      >
        <View style={styles.titleTextContainer}>
          <Text style={styles.titleText}>{props.title}</Text>
        </View>
      </ImageBackground>
    </TouchableOpacity>
  );
};

export default LifeButton;
