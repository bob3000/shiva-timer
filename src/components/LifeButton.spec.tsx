import * as React from 'react';
import * as renderer from 'react-test-renderer';
import LifeButton from './LifeButton';

describe('LifeButton', () => {
  it('snapshot test', () => {
    const tree = renderer.create(<LifeButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
