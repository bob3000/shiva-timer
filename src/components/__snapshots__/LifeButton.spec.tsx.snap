// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifeButton snapshot test 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "opacity": 0.8,
        },
        Object {
          "alignItems": "center",
          "borderColor": "#000000",
          "borderRadius": 100,
          "borderWidth": 1,
          "height": undefined,
          "justifyContent": "center",
          "opacity": 0.8,
          "width": undefined,
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "borderRadius": 40,
          "opacity": 1,
          "padding": 5,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontSize": undefined,
            "fontWeight": "bold",
          }
        }
      />
    </View>
  </View>
</View>
`;
